lib/collectors.js

```diff
@@ -0,0 +1,45 @@
+'use strict'
+
+function collectHeaders (context, queryId, response) {
+  if (!context.collectors.responseHeaders) {
+    context.collectors.responseHeaders = {}
+  }
+  context.collectors.responseHeaders[queryId] = response.headers
+}
+
+function collectStatusCode (context, queryId, response) {
+  if (!context.collectors.statusCodes) {
+    context.collectors.statusCodes = {}
+  }
+  context.collectors.statusCodes[queryId] = {
+    statusCode: response.statusCode
+  }
+}
+
+function collectExtensions (context, queryId, response) {
+  if (!context.collectors.extensions) {
+    context.collectors.extensions = {}
+  }
+  context.collectors.extensions[queryId] = response.json.extensions
```

Comment:
Here, and in the other collectors. I'll add the name of the node as well.

Something like:

context.collectors.extensions[queryId] = {service: 'user', data: response.json.extensions}

WDYT?

Reply:
I agree